### PR TITLE
Strip `!_fossa.virtual_!` from VSI paths

### PR DESCRIPTION
**Pull request by jssblck**
# Overview

When users view VSI origin paths, the `!_fossa.virtual_!` literal appended to the end of archives is confusing. These are required for the VSI backend, but not for FOSSA CLI nor for the reported results to the FOSSA web app. As such, this PR removes them when decoding the data from the VSI backend.

## Acceptance criteria

VSI results from within archives are not reported with the literal.

## Testing plan

I created an example project:
```shell
❯ zip -r cpp-vsi-demo.zip cpp-vsi-demo
❯ mv cpp-vsi-demo.zip ~/projects/scratch
❯ cabal run fossa -- analyze ~/projects/scratch -p vsi-archive-test --detect-vendored
```

This analyzed successfully and reflects the change in this PR (note that the `cpp-vsi-demo.zip` does not have the trailing literal):
```shell
* vsi analysis: succeeded
  ** /Users/jessica/projects/scratch/
  ** /cpp-vsi-demo.zip/cpp-vsi-demo/node_modules/lodash/ (locator: git+github.com/PandeoF1/px_botnet$66e6e31b39d0c770a9196c7a5443710e32367070)
  ** /cpp-vsi-demo.zip/cpp-vsi-demo/example-internal-project/vendor/facebook-folly-6695020/folly/ (locator: git+github.com/facebook/folly$2016.07.26)
  ** /cpp-vsi-demo.zip/cpp-vsi-demo/example-internal-project/vendor/libssh2/ (locator: git+github.com/libssh2/libssh2$libssh2-1.10.0)
  ** /cpp-vsi-demo.zip/cpp-vsi-demo/example-internal-project/vendor/tesseract-ocr-tesseract-cd65104/trunk/ (locator: git+github.com/tesseract-ocr/tesseract$1.03)
```

The results in FOSSA also show the origin path without the literal:
Screenshot 2023-12-13 at 1 30 12 PM

## Risks

None

## Metrics

No

## References

https://teamfossa.slack.com/archives/C0155DTGWB1/p1702410993592229
https://fossa.atlassian.net/browse/ANE-1353

## Checklist

- [x] I added tests for this PR's change (or explained in the PR description why tests don't make sense).
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [x] If this PR added docs, I added links as appropriate to the user manual's ToC in `docs/README.ms` and gave consideration to how discoverable or not my documentation is.
- [x] If this change is externally visible, I updated `Changelog.md`. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [x] If I made changes to `.fossa.yml` or `fossa-deps.{json.yml}`, I updated `docs/references/files/*.schema.json` AND I have updated example files used by `fossa init` command. You may also need to update these if you have added/removed new dependency type (e.g. `pip`) or analysis target type (e.g. `poetry`).
- [x] If I made changes to a subcommand's options, I updated `docs/references/subcommands/<subcommand>.md`.

### CODE REVIEW

File: src/App/Fossa/VSI/Types.hs [hunk 1]
```diff
@@ -77,9 +77,12 @@ instance ToJSON Locator where
...
     Locator
-      <$> obj .: "fetcher"
-      <*> obj .: "package"
-      <*> obj .: "revision"
+      <$> obj
+        .: "fetcher"
+      <*> obj
+        .: "package"
+      <*> obj
+        .: "revision"
```
Comment (jssblck) [inline src/App/Fossa/VSI/Types.hs:85]: Auto formatted

File: test/App/Fossa/VSI/TypesSpec.hs [hunk 2]
```diff
@@ -134,15 +164,20 @@ vsiTypesSpec = describe "VSI Types" $ do
...
+    generateRules' multipleInferences
+      `shouldMatchList` multipleRulesExpected
   it "Reports distinct locators for nested projects" $
-    generateRules' nestedProjectInferences `shouldMatchList` nestedProjectRulesExpected
+    generateRules' nestedProjectInferences
+      `shouldMatchList` nestedProjectRulesExpected
   it "Reports root rules correctly" $
-    generateRules' rootRuleInferences `shouldMatchList` rootRuleExpected
+    generateRules' rootRuleInferences
+      `shouldMatchList` rootRuleExpected
```
Comment (jssblck) [inline test/App/Fossa/VSI/TypesSpec.hs:180]: Auto formatted

File: src/App/Fossa/VSI/Types.hs (hunk 1 above)
Comment (csasarak) [reply src/App/Fossa/VSI/Types.hs:85]: If you use this fixities directive in `fourmolu.yaml` a better formatting should be allowed by the HLS plugin:
```
fixities:
    - infixr 0 $
    - infixl 3 <|>
    - infixr 3 &&
    - infix 4 ==
    - infixl 4 <$>, <*>
```
The relevant parts are the ones for `$`, `<$>`, `<*>` but I've added the entire directive. For this `FromJSON` instance in particular you'll have to put the formatting back manually but after that `fourmolu` shouldn't touch it. 

Sorry that this happened!

File: test/App/Fossa/VSI/TypesSpec.hs (hunk 2 above)
Comment (csasarak) [reply test/App/Fossa/VSI/TypesSpec.hs:180]: Running the formatter again with the `fixities` config I posted should switch this back to the old version.

**Review by csasarak — APPROVED** [review]
Looks good!

**Comment by jssblck** [comment]
@csasarak thanks! Much better now.

**Pull request merged**